### PR TITLE
refactor: HomeActivityViewModel Added and more lint fixes

### LINKED ISSUE #218: Refractor HomeActivity to Use ViewModel

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #218

Changes:
- Added ViewModel to Control the HomeActivity

### CODE REVIEW

**Comment by adityastic** [comment]
@aggarwalpulkit596 Review

File: app/build.gradle [hunk 1]
```diff
@@ -98,7 +98,7 @@ dependencies {
     implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.2.1'
     implementation 'androidx.browser:browser:1.0.0'
     implementation 'androidx.lifecycle:lifecycle-extensions:2.0.0'
-    implementation group: 'com.google.android.play', name: 'core', version: '1.6.0'
+    implementation 'com.google.android.play:core:1.6.0'
```
Comment (adityastic) [inline app/build.gradle]: @aggarwalpulkit596 Let me know if you agree with this. I shall remove it accordingly

**Comment by adityastic** [comment]
@aggarwalpulkit596 ok so, build issues are also fixed. you can move forward with the review now

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt [hunk 2]
```diff
@@ -400,7 +372,7 @@ class HomeActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelecte
         shortcutManager.removeAllDynamicShortcuts()
     }
 
-    fun isNetworkAvailable(context: Context): Boolean {
+    private fun isNetworkAvailable(context: Context): Boolean {
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: this function can be moved to ViewModel as well

Comment (aggarwalpulkit596) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: or you may move this function to utils class as well as delete database function

**Comment by aggarwalpulkit596** [comment]
You should always claim the bounty points after the pr is merged cause I might change bounty points depending upon the work

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt (hunk 2 above)
Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: We cannot move this to viewmodel. As viewmodel never has android imports, I shall move it to utils instead. A proper connection class

**Comment by adityastic** [comment]
@aggarwalpulkit596 great so extra fixes means extra bounty? 😅

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt (hunk 2 above)
Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: @aggarwalpulkit596 check

**Comment by aggarwalpulkit596** [comment]
Yeah sometimes extra bounty and sometimes even lesser 😅 it entirely depends upon the complexity of the work done

**Comment by adityastic** [comment]
@aggarwalpulkit596 sounds great, please review this pr so I can move forward with the next pr as well. I'll get familiar with the base till then

**Comment by adityastic** [comment]
@aggarwalpulkit596 ok so as the pr is merged, should I now claim the bounty in the boss website?

**Comment by aggarwalpulkit596** [comment]
Yup go ahead

**Comment by adityastic** [comment]
@aggarwalpulkit596 Done, Please check the claims page.

**Pull request merged**